### PR TITLE
HHH-8805 - [SchemaUpdate] javax.persistence.ForeignKey doesn't respect ConstraintMode.NO_CONSTRAINT

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-8805

### CODE REVIEW

**Comment by vladmihalcea** [comment]
@gsmet I plan on integrating this on master and think it's worth having in 5.3 as well. Let me know if you don't want it integrated there as well.

File: hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaUpdateJoinColumnNoConstraintSecondaryTableTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			"    create table Child ( " +
+			"       id bigint not null, " +
+			"        some_fk bigint, " +
+			"        primary key (id) " +
+			"    ); " +
+			" " +
+			"    create table Parent ( " +
+			"       id bigint not null, " +
+			"        primary key (id) " +
+			"    ); ";
```
Comment (gsmet) [inline hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaUpdateJoinColumnNoConstraintSecondaryTableTest.java]: Quick question as I'm not very familiar with the tests in this area: the generated scripts will be stable across the dialects? E.g. if the tests are run with MySQL/DB2/... the script will be the same?

**Comment by gsmet** [comment]
> think it's worth having in 5.3 as well

Makes totally sense. I will affect 5.3.3 as the fixVersion.

I added a small question about the tests being portable. The fix looks good.

**Comment by vladmihalcea** [comment]
@gsmet The tests are set to run on H2 only:

    @RequiresDialect(H2Dialect.class)

so, there's no possibility of a test failure on other DBs.

**Comment by gsmet** [comment]
@vladmihalcea OK perfect, then.

**Pull request merged**